test/neighborsearch.jl

```diff
@@ -0,0 +1,59 @@
+@testset "Neighborsearch" begin
+  @testset "KBallSearcher" begin
+    # KBallSearcher
+
+    #create a regular grid to test
+    pdomain = RegularGrid(10,10)
+
+    #test 1: all points should be found
+    k = 10
+    ball = BallNeighborhood(100.0)
+    kballsearcher = KBallSearcher(pdomain, k, ball)
+
+    neighbors = neighbors = Vector{Int}(undef, k)
+    xₒ = [5.0,5.0]
+    nneigh = search!(neighbors,xₒ,kballsearcher; mask=nothing)
```

Comment:
you can omit the mask argument since it defaults to nothing, it will make code cleaner and if we find a better strategy than this mask in the future the test will still work.

Reply:
also instead of testing the low-level `search!` here you could be testing the allocating version `search` that has a default implementation calling `search!`. You can just do `neighbors = search(xo, kballsearcher)`